src/dashboard/src/components/charts/bar/segmentedBarChart/SegmentedBarChart.tsx

```diff
@@ -1,76 +1,73 @@
 'use client';
 
 import { Button } from '@/components/buttons';
-import styles from './SegmentedBarChart.module.scss';
 import { Bar } from 'react-chartjs-2';
-import { defaultData } from './defaultData';
+import styles from './SegmentedBarChart.module.scss';
 
-import {
-  CategoryScale,
-  Chart as ChartJS,
-  BarElement,
-  Title,
-  Tooltip,
-  Legend,
-} from 'chart.js';
+import { useDashboardFileSystemHistoryItems } from '@/hooks/dashboard';
+import { useFiltered } from '@/store';
+import { BarElement, CategoryScale, Chart as ChartJS, Legend, Title, Tooltip } from 'chart.js';
+import React from 'react';
+import { defaultOptions } from './defaultOptions';
+import { useStorageTrends } from './useStorageTrends';
+import { extractVolumeName } from './utils';
 
 ChartJS.register(CategoryScale, BarElement, Title, Tooltip, Legend);
 
-const options = {
-  scales: {
-    x: {
-      stacked: true,
-    },
-    y: {
-      stacked: true,
-    },
-  },
-  plugins: {
-    legend: {
-      display: false
-    }
-  },
-};
+export interface ISegmentedBarChart {
+  maxVolumes?: number;
+}
 
-export const SegmentedBarChart: React.FC = () => {
-  const numDrives = 3;
-  const labelsArray = [
-    'January',
-    'February',
-    'March',
-    'April',
-    'May',
-    'June',
-    'July',
-    'August',
-    'September',
-    'October',
-    'November',
-    'December',
-  ];
+export const SegmentedBarChart = ({ maxVolumes = 4 }: ISegmentedBarChart) => {
```

Comment:
When a user selected the server in the filter it will automatically display the drives in this chart.